src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1495,6 +1495,45 @@
       (sql/do-commands
        "ALTER TABLE certnames ADD CONSTRAINT certnames_reports_id_fkey FOREIGN KEY (latest_report_id) REFERENCES reports(id) ON DELETE SET NULL")))
 
+
+(defn change-value-integer-to-numeric
+  []
+  (let [hash-type (if (sutils/postgres?) "bytea" "varchar(40)")
+        munge-hash (if (sutils/postgres?)
+                     (fn [column] (format "('\\x' || %s)::bytea" column)) identity)]
+    (sql/create-table :fact_values_transform
+                      [ "id" "bigint NOT NULL DEFAULT nextval('fact_values_id_seq')"]
+                      ["value_hash" hash-type "NOT NULL"]
+                      ["value_type_id" "bigint NOT NULL"]
+                      ["value_integer numeric(1000,0)"]
```

Comment:
Nope I'm wrong, from the manual:

```Numeric values are physically stored without any extra leading or trailing zeroes. Thus, the declared precision and scale of a column are maximums, not fixed allocations. (In this sense the numeric type is more akin to varchar(n) than to char(n).) The actual storage requirement is two bytes for each group of four decimal digits, plus three to eight bytes overhead.```

http://www.postgresql.org/docs/current/interactive/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL

Which means 1000 ~ 508 bytes. And the maximum of 131072 ~ 32776 bytes Does that sound about right?